doc/decisions/elektra_prefix.md

```diff
@@ -29,6 +32,7 @@ Rename all functions to start with elektra.
 
 - Changes in basically every application and tool, but this is automated
   with a refactoring tool @kodebach writes.
+- It is not guaranteed that the code can be compiled with every C99 compiler, because we will not always be below the 31 character limit (e.g. ELEKTRA_ERROR_VALIDATION_SYNTACTIC).
```

Comment:
Where did you read about this problem? As I understood it, the compilers might only fail if the first 31 characters are identical.

Reply:
Yes, the standard allows a compiler to ignore any character after the 31st. Therefore, if two identifiers have the same 31 character prefix, they may be treated as identical.

I will rewrite the explanation in the decision, but essentially the problem is that we would have to ensure that all our symbols have a difference in the first 31 characters. For example, we couldn't use e.g. `ELEKTRA_WARNING_VALIDATION_SYTACTIC_LEXER` and `ELEKTRA_WARNING_VALIDATION_SYTACTIC_PARSER`.